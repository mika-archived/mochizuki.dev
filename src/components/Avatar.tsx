import styled, { css } from "styled-components";

const Avatar = styled.img`
  border-radius: 999px;
  ${props => props.width && css({ width: props.width })}
  ${props => props.height && css({ height: props.height })}
`;

export default Avatar;
