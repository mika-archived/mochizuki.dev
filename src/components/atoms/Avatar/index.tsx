import styled from "styled-components";

const Avatar = styled.img`
  border-radius: 9999px;
`;

export default Avatar;
