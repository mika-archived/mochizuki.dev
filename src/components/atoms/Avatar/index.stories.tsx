import React from "react";
import { storiesOf } from "@storybook/react";

import Avatar from ".";

storiesOf("atoms/Avatar", module).add("default", () => (
  <>
    <Avatar height={128} width={128} src="https://assets.mochizuki.moe/anna.png" />
  </>
));
