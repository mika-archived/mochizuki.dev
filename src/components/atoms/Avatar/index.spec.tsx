import "@testing-library/jest-dom/extend-expect";
import "jest-styled-components";

import React from "react";
import { cleanup, render } from "@testing-library/react";

import Avatar from ".";

afterEach(cleanup);

test("does not have changes from snapshot", () => {
  const { asFragment } = render(<Avatar src="https://assets.mochizuki.moe/anna.png" />);

  expect(asFragment()).toMatchSnapshot();
});
