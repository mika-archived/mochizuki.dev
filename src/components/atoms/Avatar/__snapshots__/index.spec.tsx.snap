// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not have changes from snapshot 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 9999px;
}

<img
    class="c0"
    src="https://assets.mochizuki.moe/anna.png"
  />
</DocumentFragment>
`;
